### PR TITLE
Fix crash when brave wallet settings page is loading

### PR DESCRIPTION
**Pull request by simonhong**
|wallet_properties| could be null. All codes that references it should check
whether |error_code| is non-error and |wallet_properties| isn't null.

Still, I can't reproduce this but this pr could fix this null referencing crash as @iefremov suggested at https://github.com/brave/brave-browser/issues/2375#issuecomment-445771040

Fix https://github.com/brave/brave-browser/issues/2375

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:
`yarn test brave_unit_tests --filter=RewardsServiceTest.OnWalletProperties`

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

**Comment by bbondy** [comment]
@simonhong @NejcZdovc was this caught by previous tests that we forgot to run when this bug was introduced? Or should we add a new test to this PR to catch it?

**Comment by NejcZdovc** [comment]
@bbondy we don't have test for this code (unit test). We have browser tests, but we don't have STR for this one and it only effects old profiles, couldn't reproduce it on clean profile

**Comment by simonhong** [comment]
@NejcZdovc RewardsServiceTest.OnWalletProperties test case didn't work properly. RewardsServiceImpl::TriggerOnWalletProperties() is updated to make this test success.

File: browser/ui/webui/brave_rewards_ui.cc [hunk 1]
```diff
@@ -279,6 +279,7 @@ void RewardsDOMHandler::OnWalletProperties(
     brave_rewards::RewardsService* rewards_service,
     int error_code,
     std::unique_ptr<brave_rewards::WalletProperties> wallet_properties) {
+  DCHECK(wallet_properties);
```
Comment (NejcZdovc) [inline browser/ui/webui/brave_rewards_ui.cc]: please remove this one, because it's ok that wallet_properties is null

**Comment by NejcZdovc** [comment]
@kjozwiak @rebron @srirambv can we please uplift this to 0.58?

**Comment by kjozwiak** [comment]
@NejcZdovc uplift request to `0.58.x` approved after deliberating with @rebron 👍 Please add/remove all needed labels and ensure that the associated issue has been moved into the correct milestone.

**Comment by simonhong** [comment]
master(21806bb66219b9baa4990576c920e9ebecb2fac4)
0.60.x(a63383554b442375e8df9e72656ce275cb7280f8)
0.59.x(4c1d2ebe8473bafb487f0199104beb5579e5e145)
0.58.x(a5d3d9701134525948147efe17531f03f4cd3683)

**Pull request merged**